src/MonitoringFunctions.Windows/Incidents/Serialization/Serializer.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using MonitoringFunctions.Incidents.Models;
+using System;
+
+namespace MonitoringFunctions.Incidents.Serialization
+{
+    /// <summary>
+    /// Defines required incident serialization capabilities
+    /// </summary>
+    internal abstract class Serializer
+    {
+        /// <summary>
+        /// Creates a <see cref="string"/> representative of the specified <see cref="Incident"/> instance.
+        /// </summary>
+        /// <param name="incident">Specified <see cref="Incident"/> instance.</param>
+        /// <returns>Representative <see cref="string"/>.</returns>
+        internal string GetString(Incident incident)
+        {
+            return Serialize(incident);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="Incident"/> instance from the specified <see cref="string"/>.
+        /// </summary>
+        /// <param name="boxedIncident">Specified <see cref="string"/>.</param>
+        /// <returns>Corresponding <see cref="Incident"/> instance.</returns>
```

Comment:
Should we add some comments here about what happens if serialization fails? It looks like this will throw if JSON is malformed, but no one is handling the exception.

Reply:
the concern is to avoid throwing & improving resiliency?
agreed.
`IncidentSerializer.GetIncidentDescription` handles any serialization problem, irrespective of the provider & logs the issue.